crates/block-producer/src/batch_builder/mod.rs

```diff
@@ -33,22 +35,33 @@ pub struct BatchBuilder {
     ///
     /// Note: this _must_ be sign positive and less than 1.0.
     pub failure_rate: f32,
+    /// The batch prover to use.
+    ///
+    /// If not provided, a local batch prover is used.
+    batch_prover: BatchProver,
```

Comment:
The optional conversion happens in `new` so lets document it there only.

```suggestion
    /// The batch prover to use.
    batch_prover: BatchProver,
```